### ISSUE TITLE
Aggregated histogram_quantile produces incorrect, unstable results

### ISSUE THREAD

**Issue by lmb** [prometheus/prometheus#2598]
**What did you do?**

Aggregated a set of histograms, using

```
histogram_quantile(1, sum(irate(timeseries[3m])) by (le))
```

**What did you expect to see?**

Realistic, constant results.

**What did you see instead? Under which circumstances?**

Every evaluation of the promql generates a different result. There are spikes which go to the max bucket, even though the underlying data shows no such behaviour.

I've tried changing the resolution, etc. parameters, but that doesn't seem to make a difference. Also happens at q = 0.99, but seems more pronounced at q = 1.

Some screenshots for demonstration:

Unaggregated:
<img width="820" alt="Unaggregated histograms" src="https://cloud.githubusercontent.com/assets/368392/24797488/6080080c-1b89-11e7-9bfc-05fafa9b2f7a.png">

Aggregated, first execution:
<img width="825" alt="Aggregated, first execution" src="https://cloud.githubusercontent.com/assets/368392/24797498/6ca4336a-1b89-11e7-9d35-3c16615dc523.png">

Aggregated, second execution:
<img width="818" alt="Aggregated, second execution" src="https://cloud.githubusercontent.com/assets/368392/24797507/747e5322-1b89-11e7-9a03-72d0048bed03.png">

**Environment**

* Prometheus version: 

	Version 	1.5.2
        Revision 	16a512fe91926e2e3d0b1d2da6e7e16ceeab5f02

* client_golang version:

      334af0119a8f8fb6af5bb950d535c482cac7f836

* Prometheus configuration file:

Scrapes happen every 60s.

**Comment by brian-brazil** [comment]
This is #1893.

**Comment by lmb** [comment]
1. Issue #1887 (which apparently led to #1893) talks about graphing federated metrics. The histograms I'm aggregating are scraped by the same prometheus instance.
2. I've tried searching for this issue on the bug tracker. Seems like none of the open issues mention histograms, and all the other ones have been closed (such as this one).
3. The documentation on prometheus.io states that histograms can be aggregated. Is that a false statement?
